### ISSUE TITLE
What is the intention behind "A private, written warning from community leaders"?

### ISSUE THREAD

**Issue by ollieparanoid** [ContributorCovenant/contributor_covenant#1428]
It says in version 2.1, Enforcement Guidelines § 1 Correction:

> A **private**, written warning from community leaders

What was the intention for making a private warning at first?

We are considering to change it to "private or public" for the CoC of our community, but first we want to understand the reasoning and possibly get some feedback if that is a good idea or not. The benefit would be more intuitive for moderators if it can be done either way.

I've looked at the FAQ and git blame but didn't see anything relevant.

Thank you very much for making the Contributor Covenant!

**Comment by evolve2k** [comment]
Public shaming by community members can be it's own form of punishment. Once in play this form of punishment can be hectic and unweildy.  Rightly or wrongly, people can be judged, excluded and silently or activley punished by community members in a number of ways. 

This is an important and well established form of community power.
However once unleashed it cant be easily reversed. Even official retractions or statements that the member has now completed their sanctions, can be ignored by community members. Further secondary issues and complexities can arise where community members seek their own forms of retribution or decide to stack on to the complaint now it's clear the community member has been the recipient of an enforcement action.

In other words, be careful once you let the geanie out of the bottle, not all community members will respond the same way or with the same clarity/maturity/context as your community organisers.

There's a saying praise in public and critise in private. Issuing a private warning allows the person to 'save face', it can be a significant motivator to address their behavior as they still have not lost face publically. 

Say they slighted or upset another community member in a way that is a fit for this first level response. They could be spoken to, agree to attend training or read a certain article and even to privately apologise or have a quiet meeting, which are all approaches that can stop further behaviour from occuring and which can provide redress to key people involved.

Also by issuing a private warning the first time, it gives space for the legitimacy of further warnings and stronger sanctions the second time around. "We already spoke to you, we asked this and that, yet were back here again". This is also very helpful where there is polarisation in the community, having first undertaken a quieter approach to resolve things, future sanctions carry greater moral authority as its clear you are following a process towards justice rather than just shooting people down when they mess up.

TLDR; It's often more compassionate and just, encourages learning and makes future easier to deliver if it comes to that.

Two side comments:
If the group is tiny (say under a dozen people), where privacy is likley impossible and becomes performative, it can work that the culture is we're a small high trust group and we do all of this in the open (I'd still err to the private warning if it was me tho).

If the issue leaves the greater community in a very unresolved place either due to the act being more severe or the public nature of how the event occured it may be that you see making it public as an escalation, Id handle this as: This step is usually private and that supports quiet resolution, but on this occasion for these reasons we're making this step public, although we usually aim not to do this.

**Comment by evolve2k** [comment]
I'd like to think my response covers this off. This is more of a discussion point than a need to change the project code.
Recommend that this issue can be closed :)

**Comment by ollieparanoid** [comment]
Thank you very much for the detailed reasoning!
